ssl/record/rec_layer_s3.c

```diff
@@ -1367,6 +1369,24 @@ int ssl_set_new_record_layer(SSL_CONNECTION *s, int version,
 
     *set = OSSL_PARAM_construct_end();
 
+    /*
+     * For DTLS save off the sequence number for epoch 0 when we are setting up
+     * a new write record layer. This is needed for handling in case of HRR
+     * and we create a new write record layer for epoch 0.
+     */
+    if (direction == OSSL_RECORD_DIRECTION_WRITE
```

Comment:
Saving of the sequence value for Epoch 0. This is in case we have a hello retry request we can use the right sequence number.